src/local/document.js

```diff
@@ -0,0 +1,26 @@
+/* @flow */
+
+// The local Cozy Desktop metadata, as stored in PouchDB
+export type LocalDoc = {
+  _id: string,
+  // TODO: v3: Rename to md5sum to match remote
+  checksum?: ?string,
```

Comment:
The checksum is mandatory for files, not folders. And you can remove the mobile-specific test (the stack now generates a md5sum for files when no md5sum was given at upload, so all files in couchdb have a md5sum).